app/services/invoices/payments/adyen_service.rb

```diff
@@ -0,0 +1,178 @@
+# frozen_string_literal: true
+
+module Invoices
+  module Payments
+    class AdyenService < BaseService
+      PENDING_STATUSES = %w[AuthorisedPending Received].freeze
+      SUCCESS_STATUSES = %w[Authorised SentForSettle SettleScheduled Settled Refunded].freeze
+      FAILED_STATUSES = %w[Cancelled CaptureFailed Error Expired Refused].freeze
+
+      def initialize(invoice = nil)
+        @invoice = invoice
+
+        super(nil)
+      end
+
+      def create
+        result.invoice = invoice
+        return result unless should_process_payment?
+
+        unless invoice.total_amount_cents.positive?
+          update_invoice_payment_status(payment_status: :succeeded)
+          return result
+        end
+
+        increment_payment_attempts
+
+        adyen_result = create_adyen_payment
+
+        payment = Payment.new(
+          invoice:,
+          payment_provider_id: adyen_payment_provider.id,
+          payment_provider_customer_id: customer.adyen_customer.id,
+          amount_cents: adyen_result.dig("amount", "value"),
+          amount_currency: adyen_result.dig("amount", "currency"),
```

Comment:
Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols. (https://rubystyle.guide#consistent-string-literals)